### DISCUSSION TITLE
What's the point of setting Height and Width in NaN?

### DISCUSSION THREAD

**Discussion by SKProCH** [AvaloniaUI/Avalonia#5999] [Q&A] [answered]
I've seen several examples where the `Height` and `Width` values are set to `NaN` and the behavior changes.  
What exactly do these values do?

**Comment by maxkatz6** [comment, marked as answer]
NaN - default value for Width and Height.
If these properties had different value set (for instance from styles), and you want to set it back to default, you need to use NaN. 
It's also an equivalent of Auto value from WPF.
